cli/githubutil.py

```diff
@@ -167,9 +171,15 @@ def remove_webhooks(
     webhook_syncer = GithubWebHookSyncer(github_api)
 
     def filter_function(url):
-        concourse_id = parse_qs(urlparse(url).query).get(WebhookQueryAttributes.CONCOURSE_ID_ATTRIBUTE_NAME)
-        should_delete = concourse_id and concourse_cfg_name in concourse_id
-        return should_delete
+        parsed_url = parse_qs(urlparse(url).query)
+        concourse_id = parsed_url.get(WebhookQueryAttributes.CONCOURSE_ID_ATTRIBUTE_NAME)
+        job_mapping_id = parsed_url.get(WebhookQueryAttributes.JOB_MAPPING_ID_ATTRIBUTE_NAME)
+        url_has_matching_concourse_id = concourse_id is not None and concourse_cfg_name in concourse_id
```

Comment:
how about defining matching_job_id differently, like so:

job_id_matches = not job_id or "job_id == xxx"
should_delete = cc_id_matches and job_id_matches
return should_delete